source/log_formatter.c

```diff
@@ -51,6 +51,13 @@ static size_t s_advance_and_clamp_index(size_t current_index, int amount, size_t
 
     return next_index;
 }
+
+/* Thread-local string representation of current thread id */
+AWS_THREAD_LOCAL struct {
+    bool is_valid;
+    char repr[AWS_THREAD_ID_REPR_LEN];
```

Comment:
If you also store the actual `thread_id` here, then this becomes the thing you take the address of in aws-c-io.

Reply:
That could work, but would couple together c-io event loops and c common logging. Instead, how about changing the API for getting the current thread id to return a pointer to a thread-local `aws_thread_id` variable? The string repr can still be kept in logging, which is the only place it is needed.

```c
// in posix/thread.c
AWS_THREAD_LOCAL struct {
    bool is_valid;
    aws_thread_id thread_id;
} tl_current_thread = {.is_valid = false};

const aws_thread_id *aws_thread_current_thread_id(void) {
    if (!tl_current_thread.is_valid) {
        tl_current_thread.thread_id = pthread_self();
        tl_current_thread.is_valid = true;
    }
    return &tl_current_thread.thread_id;
}
```

Reply:
Two things:
1) "Many systems impose restrictions on the size of the thread-local memory block, in fact often rather tight limits." [wikipedia](https://en.wikipedia.org/wiki/Thread-local_storage). So it's not a great idea to make a bunch of thread-local storage variables, each of which caches one tiny thing.

2) Is there a chance of the us trying to query the tl_current_thread after the thread is gone? I know the logger usually uses one logging thread. Other threads add their statements to a queue, and the logging thread drains the queue. If a thread logs something right before it exits, then the logging thread is going to try to get its name a little bit later and it will already be gone.